scripts/finetune.py

```diff
@@ -301,13 +302,17 @@ def terminate_handler(_, __, model):
 
     LOG.info(f"Training Completed!!! Saving pre-trained model to {cfg.output_dir}")
 
+    if cfg.adapter == "lora" and cfg.relora_steps:
+        model = model.merge_and_unload()
```

Comment:
Is this necessary for only lora + relora setting? How about qlora?

Reply:
It is only necessary for lora + relora, and only when the base weights are unquantized. Thanks for poking at this, I hadn't noticed that the final save for 8-bit lora+relora was incorrect. I've made it more explicit what's going on here.